Tests/AddressURLTests/AddressURLTests.swift

```diff
@@ -108,6 +108,53 @@ final class AddressURLTests: XCTestCase {
         XCTAssertNotNil(url?.emailAddress)
         XCTAssertEqual(url?.emailAddress, "hello+tag@gorak.us")
     }
+	
+	func testValidIPv4WithPort() {
+		let validIps = [
+			"127.0.0.1:80"
+			]
+		
+		for v in validIps {
+			let vComponents = v.components(separatedBy: ":")
+			let ipAdressString = vComponents.indices.contains(0) ? vComponents[0] : nil
+			let ipPort = vComponents.indices.contains(1) ? Int(vComponents[1]) : nil
+			let ip = IPv4Address(ipAdressString ?? "")
+			XCTAssertNotNil(ip)
+			let url = URL(address: ip!).with(component: .port(ipPort))
+			XCTAssertNotNil(url)
+			XCTAssertNotNil(url!.ipv4Address)
+			XCTAssertNil(url!.ipv6Address)
+			let components = url?.dump_components()
+			XCTAssertTrue((components?.contains(where: { $0.key == "port" && $0.value as? Int == ipPort })) ?? false)
+		}
+	}
+	
+	func testValidIPv6WithPort() {
+		let validHosts = [
+			"[64:ff9b::0.0.0.0]:80",
+			"[::]:80",
+			"[::1]:80",
+			"[::ffff:0.0.0.0]:80",
+			"[::ffff:0:0.0.0.0]:80",
+			"[100::]:80",
+			"[2001::]:80",
+			"[2001:20::]:80",
+			"[2001:db8::]:80",
+			"[2002::]:80"
+		]
+		
+		for v in validHosts {
+			let vComponents = v.components(separatedBy: ":")
+			let ipAdressString = vComponents[0...vComponents.count-2].joined(separator: ":")
+			let ipPort = Int(vComponents.last ?? "")
+			let url = URLComponent.host(ipAdressString).url!.with(component: .port(ipPort))
```

Comment:
You can use `try XCTUnwrap` to avoid the `XCTAssertNotNil` test and the force unwrapping below. Using `XCTUnwrap` may cleanup some of this code with less optionals and make things easier to read. 

Thanks

https://sarunw.com/posts/unwrap-optional-values-in-xctest-with-xctunwrap/